rust_book_ex_8_3/src/vector_ex.rs

```diff
@@ -0,0 +1,28 @@
+use std::collections::HashMap;
+
+pub fn get_mean(v: Vec<u32>) -> f32{
+    let sum: u32 = v.iter().sum();
+    return sum as f32 / v.len() as f32;
+}
+
+pub fn get_median(v: Vec<u32>) -> u32 {
+    let size = v.len();
+    return v[size / 2];
+}
+
+pub fn get_mode(v: Vec<u32>) -> u32 {
```

Comment:
Taking a vector by value here is wrong. You're basically forcing your caller to move their vector into this function - so they will be forced to `.clone()` it if they want to use it later.

A better alternative would be `&Vec<u32>`; and a better yet alternative would be a slice: `&[u32]` since many other contiguous containers can dereference to it.